cmd/kube-apiserver/app/server.go

```diff
@@ -119,6 +122,10 @@ cluster's shared state through which all other components interact.`,
 				// if we are running openshift, we modify the admission chain defaults accordingly
 				admissionenablement.InstallOpenShiftAdmissionPlugins(s)
 
+				if err := openshiftfeatures.InitializeFeatureGates(feature.DefaultMutableFeatureGate, configv1.FeatureGateRouteExternalCertificate); err != nil {
```

Comment:
Why not move this to `ConfigToFlags` or, better `ForceGlobalInitializationForOpenShift`?

Reply:
I added the feature gate initialization to `ForceGlobalInitializationForOpenShift` func.
Not a problem, except I needed to return an error in the same func so refactored accordingly.

Reply:
@vrutkovs The e2e(s) were failing because of the move of the `openshiftfeatures.InitializeFeatureGates` func call inside of `ForceGlobalInitializationForOpenShift`. Also, I tried debugging the same a bit further, even moving it to `ConfigToFlags` causes the same error.

```
E0205 21:57:21.034564       1 run.go:74] "command failed" err="cannot add a feature gate after adding it to the flag set"
```
The same is causing the apiserver to not come up on the bootstrap node i.e. failing all e2e(s).

Reply:
To be precise, we do need the `openshiftfeatures.InitializeFeatureGates` for openshift-kube-apiserver to be able to add the openshift-specific feature gates on top of what comes from k8s upstream. 

The exact reason why this issue is caused is because at runtime, `f.AddFlag()` is getting called before our `f.Add()` (from feature_gates: https://github.com/openshift/kubernetes/blob/HEAD/staging/src/k8s.io/component-base/featuregate/feature_gate.go) call inside of our `InitializeFeatureGates` func call. The first call to `f.AddFlag()` sets `f.closed` to `true` and subsequent calls to `f.Add()` errors out.

This is because, `f.AddFlag()` was called inside of:
```
runtime/debug.Stack()
        /home/swghosh/go121/go/src/runtime/debug/stack.go:24 +0x5e
k8s.io/component-base/featuregate.(*featureGate).AddFlag(0xc000307a00, 0x5dd29c0?)
        vendor/k8s.io/component-base/featuregate/feature_gate.go:325 +0x2f
k8s.io/apiserver/pkg/server/options.(*ServerRunOptions).AddUniversalFlags(0xc000c91800, 0x548e0fc?)
        vendor/k8s.io/apiserver/pkg/server/options/server_run_options.go:361 +0x426
k8s.io/kubernetes/pkg/controlplane/apiserver/options.(*Options).AddFlags(0xc00076a360, 0x88e75c8?)
        pkg/controlplane/apiserver/options/options.go:129 +0x54
k8s.io/kubernetes/cmd/kube-apiserver/app/options.(*ServerRunOptions).Flags(0xc00085a2c0)
        cmd/kube-apiserver/app/options/options.go:102 +0x5f
k8s.io/kubernetes/cmd/kube-apiserver/app.NewAPIServerCommand()
        cmd/kube-apiserver/app/server.go:172 +0x105
main.main()
        cmd/kube-apiserver/apiserver.go:33 +0x13
```
i.e. exactly inside of [cmd/kube-apiserver/app/server.go:172](https://github.com/openshift/kubernetes/blob/2f773e8d16da18d91d717cb7e1aca3d000396e2d/cmd/kube-apiserver/app/server.go#L172#L172).

Reply:
Added the fix in https://github.com/openshift/kubernetes/pull/1852/commits/c81317550afddc99662f707469cb1497b2028e87

Reply:
@soltysh In this thread is illustrated what happens when I really move the 

```go
        if err := features.InitializeFeatureGates(feature.DefaultMutableFeatureGate,
		OpenShiftKubeAPIServerFeatureGates...,
	); err != nil {
		klog.Fatal(err)
	}
```

block of code into `openshift-kube-apiserver/openshiftkubeapiserver/flags.go` inside of `ConfigToFlags(...)`.